### DISCUSSION TITLE
Error on creating connection via http://localhost:8006/v1/connections api if destination is a custom connector

### DISCUSSION THREAD

**Discussion by abhijithcdreamorbit** [airbytehq/airbyte#35724] [Connector Questions]
I am trying to create connection for two custom connectors using (**http://localhost:8006/v1/connections**)
But the api gives error saying (**Submitted value could not be found: null**)
{
  "name": "test-connection",
  "sourceId": "7fe9f226-b794-4b2e-801d-0021bfc8ac71",
  "destinationId": "1d7c4aa8-4afd-473d-895a-aa925a53b90c",
  "workspaceId": "7f1b5b55-a20a-4485-b8c9-a75b7ba6c749",
  "status": "active",
  "configurations": {
                "streams": [
                    {
                        "name": "stream1",
                        "syncMode": "incremental_append",
                        "cursorField": [
                            "last_modified_date_time"
                        ]
                    },
                    {
                        "name": "stream2",
                        "syncMode": "incremental_append",
                        "cursorField": [
                            "last_modified_date_time"
                        ]
                    }
                ]
            },
  "schedule": {
    "scheduleType": "manual"
  },
  "dataResidency": "auto",
  "nonBreakingSchemaUpdatesBehavior": "ignore",
  "namespaceDefinition": "destination",
  "namespaceFormat": "${SOURCE_NAMESPACE}"
}

Additionally, I did try by giving an empty stream array and it worked. (Giving an impression that its failing to process streams config)
I also tried from UI. And was able to create a connection and sync. So no issues with how the connectors are built

Also this doesn't seem to work only when the destination is a custom connector. (As I tried with an inbuilt destination connector **E2E Testing** and that worked with **same payload** with **E2E destinationId**)

Looking at the code
https://github.com/airbytehq/airbyte-platform/blob/main/airbyte-api-server/src/main/kotlin/io/airbyte/api/server/services/DestinationService.kt#L285C7-L285C40
I see that It seems to fail while trying to fetch **destinationDefinitionId** from the mapper (Ideally it shouldn't be fetching from a mapper as its a custom connector)

**Comment by mn-mikke** [comment]
I have experienced the same problem.

**Comment by tinomerl** [comment]
just wanted to leave this here in case anyone else is still experiencing this: The fix for me was updating my airbyte instance.

[a comment by an automated account is not shown]
